File: Lab2/src/EncryptionContext.java
public class EncryptionContext {
    private EncryptionStrategy strategy;


    public void setEncryptionStrategy(EncryptionStrategy strategy) {
        this.strategy = strategy;
    }

    public byte[] encrypt(String data, String key) throws Exception {
        return strategy.encrypt(data, key);
    }

    public String decrypt(byte[] encryptedData, String key) throws Exception {
        return strategy.decrypt(encryptedData, key);
    }
}

File: Lab_2_EnDcypt/Lab_2_EnDcypt/EncryptionContext.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab_2_EnDcypt
{
    public class EncryptionContext
    {
        private IEncryptionStrategy _encryptionStrategy;

        public void SetEncryptionStrategy(IEncryptionStrategy encryptionStrategy)
        {
            _encryptionStrategy = encryptionStrategy;
        }

        public string Encrypt(string plainText)
        {
            return _encryptionStrategy.Encrypt(plainText);
        }

        public string Decrypt(string cipherText)
        {
            return _encryptionStrategy.Decrypt(cipherText);
        }
    }

}
